app/vmalert/remotewrite/debug_client.go

```diff
@@ -0,0 +1,266 @@
+package remotewrite
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/golang/snappy"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/utils"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promauth"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/prompbmarshal"
+)
+
+// DebugClient won't push series periodically, but have debug flush
+type DebugClient struct {
+	addr         string
+	c            *http.Client
+	authCfg      *promauth.Config
+	input        chan prompbmarshal.TimeSeries
+	maxBatchSize int
+	maxQueueSize int
+
+	wg     sync.WaitGroup
+	doneCh chan struct{}
+
+	debugflushWG sync.WaitGroup
+	debugflushCh chan struct{}
+}
+
+// NewDebugClient returns a debug remotewrite client
+func NewDebugClient(ctx context.Context) (*DebugClient, error) {
+	if *addr == "" {
+		return nil, nil
+	}
+
+	t, err := utils.Transport(*addr, *tlsCertFile, *tlsKeyFile, *tlsCAFile, *tlsServerName, *tlsInsecureSkipVerify)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create transport: %w", err)
+	}
+
+	authCfg, err := utils.AuthConfig(
+		utils.WithBasicAuth(*basicAuthUsername, *basicAuthPassword, *basicAuthPasswordFile),
+		utils.WithBearer(*bearerToken, *bearerTokenFile),
+		utils.WithOAuth(*oauth2ClientID, *oauth2ClientSecret, *oauth2ClientSecretFile, *oauth2TokenURL, *oauth2Scopes),
+		utils.WithHeaders(*headers))
+	if err != nil {
+		return nil, fmt.Errorf("failed to configure auth: %w", err)
+	}
+	c := &DebugClient{
+		c: &http.Client{
+			Timeout:   *sendTimeout,
+			Transport: t,
+		},
+		addr:         strings.TrimSuffix(*addr, "/"),
+		authCfg:      authCfg,
+		input:        make(chan prompbmarshal.TimeSeries, defaultMaxQueueSize),
+		maxBatchSize: *maxQueueSize,
+		maxQueueSize: *maxQueueSize,
+		doneCh:       make(chan struct{}),
+
+		debugflushCh: make(chan struct{}, 1),
+	}
+	c.run(ctx)
+	return c, nil
+}
+
+// Push adds timeseries into queue for writing into remote storage.
+// Push returns and error if client is stopped or if queue is full.
+func (c *DebugClient) Push(s prompbmarshal.TimeSeries) error {
```

Comment:
We use `push` to store single ts to input channel, and flush `[]TimeSeries` in one request now. It seems more efficient to me.

Reply:
Would be nice to verify how much slower it becomes if implemented as I suggested.
I understand it could be slower, but I don't like the complexity of the DebugClient. I believe, it could be much simpler and predictable as it is right now.